mods/accent-color-sync.wh.cpp

```diff
@@ -172,225 +165,264 @@ BOOL set_DWORD(std::wstring sk, std::wstring v, DWORD data)
-        set_DWORD(L"SOFTWARE\\StartIsBack", L"TaskbarAlpha", settings.isTransparent ? newSettings.nIntensity / 100.0 * 255.0: 255.0);
+        int opacity = (settings.opacity * 0.01) * (217.0 - 26.0) + 26.0;
+
+        getColorBalances
+        (
+            opacity,
+            colour,
+            afterglow,
+            blur
+        );
+        
+        // Changing the ColorizationColor and Afterglow also affects the intensity slider
     }
+    
+    // *********************************************
+    // Actually do the registry editing
+    // *********************************************
+    set_DWORD(dwmKey, L"og_ColorizationColorBalance", colour);
+    set_DWORD(dwmKey, L"og_ColorizationAfterglowBalance", afterglow);
+    set_DWORD(dwmKey, L"og_ColorizationBlurBalance", blur);
 
-    RefreshDWM();
-    return TRUE;
+    // Other registry values
+    if (exists_DWORD(dwmKey, L"GlassOpacity")) set_DWORD(dwmKey, L"GlassOpacity", 0);
+
+    PostMessage(FindWindow(TEXT("dwm"), nullptr), WM_DWMCOLORIZATIONCOLORCHANGED, 0, 0);
 }
 
-BOOL isChangeable(bool force)
-{
-    if (isWorking && !force) return FALSE;
+#pragma region ## DirectUI hooks ##
+typedef ATOM WINAPI (*StrToId_T)(unsigned const short*);
+StrToId_T StrToID;
+
+// Pointers to DUI elements
+intptr_t intensitySlider = 0;
+
+typedef unsigned short(*__cdecl Element_GetID_T)(class Element*, void*);
+Element_GetID_T Element_GetID;
 
-    current = tempSettings.clrColor;
-    DwmGetColorizationParameters(&tempSettings);
+typedef void(* __cdecl Element_OnPropertyChanged_T)(class Element*, class PropertyInfo const *,int,class Value *,class Value *);
+Element_OnPropertyChanged_T Element_OnPropertyChanged;
 
-    if (tempSettings.clrColor == current)
+void __cdecl Element_OnPropertyChanged_hook(class Element* This, class PropertyInfo const *prop, int integer, class Value *valueA, class Value *valueB)
+{
+    Element_OnPropertyChanged(This, prop, integer, valueA, valueB);
+
+    ATOM id = Element_GetID(This, &This);
+    if (intensitySlider != id && id == StrToID((unsigned const short*)L"IntensitySlider"))
     {
-        return FALSE;
+        intensitySlider = reinterpret_cast<intptr_t>(This);
     }
+}
 
-    if (current != 0 || force)
-    {
-        isWorking = true;
+long(*__cdecl CCTrackBar_SetThumbPosition)(class CCTrackBar*, int);
+long __cdecl CCTrackBar_SetThumbPosition_hook(class CCTrackBar* This, int value)
+{
+    intptr_t current = reinterpret_cast<intptr_t>(This);
 
-        /* Each unconverted color value contains the opacity in AArrggbb */
-        return TRUE;
+    // Track bar value
+    if (current > 0 && current == intensitySlider)
+    {
+        settings.opacity = (value - 10) / 75.0 * 100.0;
+        WriteNewColor();
     }
 
-    return FALSE;
+    return CCTrackBar_SetThumbPosition(This, value);
 }
+#pragma endregion
 
-void ColorFunction(BOOL force = false)
-{
-    if (!isWorking) 
+WindhawkUtils::SYMBOL_HOOK hooks[] = {

```

Comment:
Have you seen the warning for this line? I added a new rule for naming symbol hook variables, to be able to optimize loading symbols. Let me know if the message is not clear enough.

Reply:
I saw it, but actually didn't understand what it was supposed to mean at first until after scanning it now